### PR TITLE
remove component related code from native

### PR DESCRIPTION
**Pull request by stanleyljl**
Re: #

### Changelog

* remove component related code from native

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by jk20012001 — APPROVED** [review]

File: native/cocos/gi/light-probe/LightProbe.cpp [hunk 1]
```diff
@@ -209,6 +211,106 @@ void LightProbeInfo::clearSHCoefficients() {
...
+bool LightProbeInfo::addNode(Node* node) {
+    if (!node) {
+        return false;
+    }
+
+    for (auto &item : _nodes) {
+        if (item.node == node) {
+            return false;
+        }
+    }
```
Comment (dumganhar) [inline native/cocos/gi/light-probe/LightProbe.cpp:227]: Could use 
```
if (std::find(_nodes.begin(), _node.end(), node) != _node.end())  {
     return false;
}
```
?

File: native/cocos/gi/light-probe/LightProbe.cpp [hunk 2]
```diff
@@ -209,6 +211,106 @@ void LightProbeInfo::clearSHCoefficients() {
...
+
+void LightProbeInfo::update(bool updateTet) {
+    ccstd::vector<Vec3> points;
+
+    for (auto &item : _nodes) {
+        auto *node = item.node;
+        auto &probes = item.probes;
+        const auto &worldPosition = node->getWorldPosition();
+
+        for (auto &probe : probes) {
```
Comment (dumganhar) [inline native/cocos/gi/light-probe/LightProbe.cpp:266]: What could the size of `points` be?  Should we reserve its capacity to avoid memory re-allocation?

Comment (dumganhar) [reply native/cocos/gi/light-probe/LightProbe.cpp:266]: Will the `update` method be invoked frequently?

Comment (stanleyljl) [reply native/cocos/gi/light-probe/LightProbe.cpp:266]: Only invoked when user click the generate probe button

**Review by dumganhar — APPROVED** [review]

**Pull request merged**